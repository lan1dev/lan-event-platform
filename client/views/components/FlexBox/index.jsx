import styled from 'styled-components';

/* eslint-disable react/destructuring-assignment */
const FlexBox = styled.div`
  display: flex;
  height: ${props => (props.height ? props.height : 'auto')};
  flex-direction: ${props => (props.direction ? props.direction : 'row')};
  justify-content: ${props => (props.justify ? props.justify : 'flex-start')};
  align-items: ${props => (props.align ? props.align : 'flex-start')};
`;
/* eslint-enable react/destructuring-assignment */

export default FlexBox;
