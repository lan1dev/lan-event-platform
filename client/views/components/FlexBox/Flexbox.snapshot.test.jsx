import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';
import FlexBox from './index';

describe('FlexBox', () => {
  let props;
  let wrapper;

  const Wrapper = () => {
    if (!wrapper) {
      wrapper = shallow(<FlexBox {...props}>{props.children}</FlexBox>);
    }
    return wrapper;
  };

  // Reset Component and Props before each test.
  beforeEach(() => {
    props = {};
    wrapper = undefined;
  });

  describe('without props', () => {
    it('Should match snapshot', () => {
      expect(toJson(Wrapper())).toMatchSnapshot();
    });
  });

  describe('with direction prop', () => {
    it('Should match snapshot', () => {
      props = {
        direction: 'column'
      };
      expect(toJson(Wrapper())).toMatchSnapshot();
    });
  });

  describe('with jusify prop', () => {
    it('Should match snapshot', () => {
      props = {
        justify: 'center'
      };
      expect(toJson(Wrapper())).toMatchSnapshot();
    });
  });

  describe('with align prop', () => {
    it('Should match snapshot', () => {
      props = {
        align: 'center'
      };
      expect(toJson(Wrapper())).toMatchSnapshot();
    });
  });
});
