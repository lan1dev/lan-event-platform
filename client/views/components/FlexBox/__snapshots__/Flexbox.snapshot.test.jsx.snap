// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlexBox with align prop Should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  className="c0"
/>
`;

exports[`FlexBox with direction prop Should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

<div
  className="c0"
  direction="column"
/>
`;

exports[`FlexBox with jusify prop Should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

<div
  className="c0"
/>
`;

exports[`FlexBox without props Should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

<div
  className="c0"
/>
`;
